HW 17 send to review

Ximelz opened the pull request.

Ximelz committed:
HW 17 send to review

Ximelz committed:
Изменена строка сообщения текущих задач пользователя на основании исправления условия ДЗ.

PugachA commented on Otus_Notification_Homework_17/Otus_Notification_Homework_17/BackgroundTasks/NotificationBackgroundTasks/DeadlineBackgroundTask.cs:
```diff
@@ -0,0 +1,36 @@
+            this.toDoRepository = toDoRepository;
+        }
+
+        private INotificationService notificationService;
```
readonly

PugachA commented on Otus_Notification_Homework_17/Otus_Notification_Homework_17/BackgroundTasks/NotificationBackgroundTasks/NotificationBackgroundTask.cs:
```diff
@@ -0,0 +1,33 @@
+            this.notificationService = notificationService;
+        }
+
+        private INotificationService notificationService;
```
readonly

PugachA commented on Otus_Notification_Homework_17/Otus_Notification_Homework_17/BackgroundTasks/NotificationBackgroundTasks/TodayBackgroundTask.cs:
```diff
@@ -0,0 +1,38 @@
+            this.toDoRepository = toDoRepository;
+        }
+
+        private INotificationService notificationService;
```
readonly

PugachA commented on Otus_Notification_Homework_17/Otus_Notification_Homework_17/Core/Services/NotificationService/NotificationService.cs:
```diff
@@ -0,0 +1,72 @@
+        {
+            this.factory = factory;
+        }
+        private IDataContextFactory<ToDoDataContext> factory;
```
readonly

PugachA commented on Otus_Notification_Homework_17/Otus_Notification_Homework_17/Infrastructure/DataAccessDb/ToDoDataContext.cs:
```diff
@@ -29,5 +30,11 @@ public ToDoDataContext(string connectionString) : base(ProviderName.PostgreSQL,
+        /// <summary>
+        /// Выгруженная таблица нотификаций.
+        /// </summary>
+        public ITable<NotificationModel> NotificationTable => this.GetTable<NotificationModel>();
```
Лучше назвать Notifications. Аналогично другие свойства

PugachA commented on Otus_Notification_Homework_17/Otus_Notification_Homework_17/Infrastructure/DataAccessDb/Repositories/SqlToDoRepository.cs:
```diff
@@ -151,5 +151,24 @@ public async Task<IReadOnlyList<ToDoItem>> GetActiveByUserId(Guid userId, Cancel
+                             .Where(x => x.UserId == userId)
+                             .Where(y => y.State == ToDoItemState.Active)
+                             .Where(x => x.DeadLine >= from && x.DeadLine < to)
+                             .ToList()
```
ToListAsync(ct) Работать с БД нужно асинхронно

PugachA commented on Otus_Notification_Homework_17/Otus_Notification_Homework_17/Infrastructure/DataAccessFiles/FileToDoRepository.cs:
```diff
@@ -222,5 +223,14 @@ private List<ToDoItem> GetTasksByUserId(Guid userId)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            return GetActiveByUserId(userId, ct).Result
```
Нужно разделить GetActiveByUserId и Where чтобы использовать await

PugachA commented on Otus_Notification_Homework_17/Otus_Notification_Homework_17/Infrastructure/DataAccessFiles/FileUsersRepository.cs:
```diff
@@ -58,5 +59,26 @@ public Task Add(ToDoUser user, CancellationToken ct)
             return Task.CompletedTask;
         }
+
+        public async Task<IReadOnlyList<ToDoUser>> GetUsers(CancellationToken ct)
```
Поддерживать старые репозитории особо нет смысла, так как они не используются

Ximelz committed:
Внесены изменения, связанные с 15 и 16 ДЗ

Ximelz committed:
Внес изменения на основании PR:
1. Добавил индекс для UserId в таблицу нотификаций.
2. Добавил асинхронную работу с БД и асинхронными методами.
3. К неизменяемым полям добавил модификатор readonly.

Ximelz committed:
Добавление токена отмены в асинхронные методы при работе с БД.
